lms/djangoapps/certificates/models.py

```diff
@@ -118,6 +122,17 @@ def certificate_for_student(cls, student, course_id):
         return None
 
 
+@receiver(post_save, sender=GeneratedCertificate)
+def handle_post_cert_generated(sender, instance, **kwargs):  # pylint: disable=no-self-argument, unused-argument
+    """
+    Handles post_save signal of GeneratedCertificate, and mark user collected
+    course milestone entry if user has passed the course
+    or certificate status is 'generating'.
+    """
+    if settings.FEATURES.get('ENABLE_PREREQUISITE_COURSES') and instance.status == CertificateStatuses.generating:
+        fulfill_course_milestone(instance.course_id, instance.user)
+
+
```

Comment:
@mattdrayer seems like the fulfillment PR got merged into this branch, given that I see this here...